### ISSUE TITLE
CopyQ main window in macOS not respecting "hide_main_window_in_task_bar" setting

### ISSUE THREAD

**Issue by jromero2k** [hluk/CopyQ#1381]
**Describe the bug**
Even if hide_main_window_in_task_bar is set to false, CopyQ doesn't show it in the taskbar. Doing something and switching back to CopyQ without the mouse is impossible 

**To Reproduce**
1. `copyq config hide_main_window_in_task_bar 0`
2. Exit CopyQ (just to make sure)
3. `copyq config hide_main_window_in_task_bar`
4. Still same (wrong) behavior

**Expected behavior**
When `copyq config hide_main_window_in_task_bar 0` the main window should show in the task switcher and I should be able to activate it without clicking on Mission Control or Show/Hide twice

**Screenshots**
-

**Version, OS and Environment**
CopyQ Clipboard Manager v3.11.0
Qt: 5.13.2
Compiler: GCC
Arch: x86_64-little_endian-lp64
OS: macOS Mojave (10.14)

**Additional context**
-

**Comment by orschiro** [comment]
I am not sure if related, but whenever I use the shortcut to show the tray menu, it shows the main window as well. I cannot see the same behaviour happen on Linux.

**Comment by hluk** [comment]
It's a bit complicated because there are three options that affect the behavior:

- `hide_main_window_in_task_bar` - if enabled, main window should never show up in task bar
- `hide_main_window` - if enabled, main window will hide instead of minimizing
- `disable_tray` - if enabled, tray icon won't show up and main window will be minimized to task bar (unless `hide_main_window` is enabled)

**Comment by orschiro** [comment]
Thanks! Let me play with them.

**Comment by hluk** [comment]
@orschiro Actually, showing main window when tray menu is requested is probably unrelated to this. Can you create new issue for that? Thanks.

**Comment by orschiro** [comment]
OK, will do, thanks!

**Comment by jromero2k** [comment]
Any progress with this?

**Comment by jromero2k** [comment]
Today I installed CopyQ 3.13 and it's still the same mess.. there's no way to show the main window. 

This version change log says something about the app's menu but the menu is (still) only showing if I bring up the Preferences dialog (but since it's modal, there's nothing meaningful to click..)

**Comment by hluk** [comment]
To show the app in task bar, you also have to enable `disable_tray` option:

```bash
copyq config disable_tray 1
copyq config hide_main_window_in_task_bar 0
copyq config hide_main_window 0
# in some cases, the following command could be required to update the state in task bar
copyq show
```

If you close the main window, it would only minimize to task bar. Not sure if the behavior is different on macOS.

> This version change log says something about the app's menu but the menu

If you mean the new `native_menu_bar` option, that is only related to the application menu bar (#1444) - i.e. File, Edit menus etc.

**Comment by jromero2k** [comment]
Thanks! I had already tried those options.. I tried again, and even restarted the app. 

Results: I still can't access CopyQ, the app never shows when switching/cycling between apps (Cmd+TAB in my case).

After disabling the tray, (predictably) I can't see the latest clipboard items. Reenabling it showed the icon again but there's no tray context menu, clicking the icon just toggles the main window instead of the previous behavior.

> If you mean the new native_menu_bar option, that is only related to the application menu bar (#1444) - i.e. File, Edit menus etc.

Yes, I meant the application menu bar (which in macOS stays visible globally and switches depending on which app is focused.) In the case of CopyQ it doesn't update, keeps showing the menu of the previously active app. If I set `native_menu_bar` to `false`, at least the menu shows within the main window, but if set to `true` it doesn't show anywhere. So with `false` it kind of works, but it's ugly and not the native expected behavior.

**Comment by hluk** [comment]
> Results: I still can't access CopyQ, the app never shows when switching/cycling between apps (Cmd+TAB in my case).

I was mainly talking about app icon in task bar but it normally should show up in the switcher also ... unless the window is hidden.

Is the application window open when you use Cmd+TAB? Maybe minimized?

BTW, you can use `copyq config close_on_unfocus 0` for the main window to stop hiding when it loses focus.

> clicking the icon just toggles the main window instead of the previous behavior.

This was changed in the new version. If you want the previous behavior, use `copyq config tray_menu_open_on_left_click 1`.

> Yes, I meant the application menu bar (which in macOS stays visible globally and switches depending on which app is focused.) In the case of CopyQ it doesn't update, keeps showing the menu of the previously active app.

Unfortunately, I think it's problem in Qt GUI framework missing some support for more recent macOS versions.

You can try test build with newer Qt version: https://github.com/hluk/CopyQ/releases/tag/macos-window

If that works, I can rebuild the latest app with newer Qt. It may actually fix some issues with window handling too.

**Comment by jromero2k** [comment]
> Is the application window open when you use Cmd+TAB? Maybe minimized?

Nope. Open and focused, or any other combination, CopyQ just doesn't show there. It does show in Mission Control.

> BTW, you can use `copyq config close_on_unfocus 0` for the main window to stop hiding when it loses focus.

It's already `false`:

```
~$ copyq config close_on_unfocus
false
```

> This was changed in the new version. If you want the previous behavior, use `copyq config tray_menu_open_on_left_click 1`.

Yes. That fixed it, thanks. I don't remember seeing that in the release notes, but may be I wasn't paying enough attention.

> You can try test build with newer Qt version: https://github.com/hluk/CopyQ/releases/tag/macos-window

OK, I'll try later and report back :)
